addons/website/tests/test_performance.py

```diff
@@ -274,11 +274,13 @@ def test_30_perf_sql_queries_page_no_layout(self):
             'website_page': 2,
             # 1. `_serve_page` search page matching URL..
             # 2. ..then reads it (`is_visible`)
+            'website': 1,
+            # Check if website.cookies_bar is active
```

Comment:
Why was it not the case before? :thinking:

Reply:
This is due to my call in `_prepare_frontend_environment` to `_is_allowed_cookie('optional')` (for caching reasons), which itself checks `get_current_website().cookies_bar`. (So now that I re-check that, I suppose the query is in fact the one from `get_current_website`.)

Reply:
Mmh, ok, and this test does not check for the queries made during the view rendering then? I don't really know what surrounds all those perf tests :grimacing: cc @rdeodoo ?

Reply:
I wouldn't mind too much about this test, unless it really gets out of control.
It's testing an (almost) unexisting case of a website.page not doing a t-call to the layouts. It's typically the case of advanced .scss / .xml / .js pages which no one knows about anyway.

What we would do if we really wanted to get rid of that query is to add `cookies_bar` into the `webite`'s `_get_cached_values` method.

> this test does not check for the queries made during the view rendering then? I don't really know what surrounds all those perf tests

It tests exactly how many SQL queries are needed for a client call to render a page, so it's including rendering and business code.
It's basically the number you see in your logs next a `GET` query. The number in the test should match the number in your log in real case.

The `cache=False` is basically useless, I kept it for no good reason, just that it was written so it has small value as it prevents cold requests to skyrocket, but we don't care that much about cold state.

Note that it's actually t-cache which is disabled, only, not orm cache or anything. Initially I made a mistake when I introduced the perf tests, it was actually testing always in t-cache disabled mode.. Which was not what we wanted at all. So I fixed it by splitting the lines in 2, but I could also have just remove the cache=False one.

So at the end, for this case, it's just that without this change we don't need to ever query the website table when rendering the page, it's probably because due to the fact that we don't render the layout, we never read a website value which is not part of `_get_cached_values()`.

Reply:
> It's testing an (almost) unexisting case of a website.page not doing a t-call to the layouts.

Oh, yes, that's the answer I needed to understand in the end, thanks :+1: No t-call, no cookie bar, no call to website. It's ok it now will.